### ISSUE TITLE
Epoch error

### ISSUE THREAD

**Issue by theheking** [dpeerlab/spectra#50]
Currently fitting the spectra model using use_cell_type=True, which runs successfully on 2 epochs. But leads to all NA output with epoch # > 20. Not sure what would be driving the error here. 

`# fit the model (We will run this with only 2 epochs to decrease runtime in this tutorial)
model = spc.est_spectra(adata=adata, 
    gene_set_dictionary=annotations, 
    use_highly_variable=True,
    cell_type_key="blueprint_labels_mapped", 
    use_weights=True,
    lam=0.1, # varies depending on dsata and gene sets, try between 0.5 and 0.001
    delta=0.001, 
    kappa=None,
    rho=0.001, 
    use_cell_types=True,
    n_top_vals=50,
    label_factors=True, 
    overlap_threshold=0.2,
    clean_gs = True, 
    min_gs_num = 3,
    num_epochs=100 #here running only 2 epochs for time reasons, we recommend 10,000 epochs for most datasets
)`

`Cell type labels in gene set annotation dictionary and AnnData object are identical
Your gene set annotation dictionary is now correctly formatted.
`

![Screenshot 2024-11-28 at 2 53 17 PM](https://github.com/user-attachments/assets/891c80d1-fc7b-4f47-b936-359b598ce6ff)

![image](https://github.com/user-attachments/assets/4660c2b6-edd3-473d-bfdc-4adfeb448c18)
